From: Ben Hutchings <ben@decadent.org.uk>
Date: Sun, 18 Feb 2018 01:26:17 +0000
Subject: block: Avoid ABI change in 3.16.54
Forwarded: not-needed

Commit 5acb3cc2c2e9 ("blktrace: Fix potential deadlock between delete
& sysfs ops") added a new member in the middle of struct
request_queue.  This structure is always allocated by built-in code,
and the new member is only used by built-in code, so we can add the
new member at the end without an ABI change.  So move it to the end
and hide it from genksyms.

---
--- a/include/linux/blkdev.h
+++ b/include/linux/blkdev.h
@@ -448,7 +448,6 @@ struct request_queue {
 	int			node;
 #ifdef CONFIG_BLK_DEV_IO_TRACE
 	struct blk_trace	*blk_trace;
-	struct mutex		blk_trace_mutex;
 #endif
 	/*
 	 * for flush operations
@@ -489,6 +488,12 @@ struct request_queue {
 
 	struct blk_mq_tag_set	*tag_set;
 	struct list_head	tag_set_list;
+
+#ifndef __GENKSYMS__
+#ifdef CONFIG_BLK_DEV_IO_TRACE
+	struct mutex		blk_trace_mutex;
+#endif
+#endif
 };
 
 #define QUEUE_FLAG_QUEUED	1	/* uses generic tag queueing */
